client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/InteractionSection/InteractionEventForm.tsx

```diff
@@ -0,0 +1,235 @@
+import InteractionEventDialogFields from 'models/enums/InteractionsEventDialogContext/InteractionEventDialogFields';
+
+import AddressForm from './AddressForm/AddressForm';
+import useStyles from './InteractionEventFormStyles';
+import PlaceNameForm from './PlaceNameForm/PlaceNameForm';
+import BusinessContactForm from './BusinessContactForm/BusinessContactForm';
+
+export const defaultContact: Contact = {
+    firstName: '',
+    lastName: '',
+    phoneNumber: '',
+    idNumber: '',
+    contactType: -1,
+    creationTime: new Date(),
+    involvedContact: null,
+};
+
+const InteractionEventForm: React.FC<InteractionEventFormProps> = (
+    { onPlaceSubTypeChange, isVisible, interactionData, isNewInteraction }: InteractionEventFormProps): JSX.Element => {
+    const {control, watch, clearErrors, setValue, errors, setError} = useFormContext();
+
+    const placeType = watch(InteractionEventDialogFields.PLACE_TYPE);
+    const placeSubType = watch(InteractionEventDialogFields.PLACE_SUB_TYPE);
+    const interactionStartTime = watch(InteractionEventDialogFields.START_TIME);
+    const interactionEndTime = watch(InteractionEventDialogFields.END_TIME);
+    const isUnknownTime = watch(InteractionEventDialogFields.UNKNOWN_TIME);
+    const locationAddress = watch(InteractionEventDialogFields.LOCATION_ADDRESS);
+    const placeName = watch(InteractionEventDialogFields.PLACE_NAME);
+    const placeDescription = watch(InteractionEventDialogFields.PLACE_DESCRIPTION);
+
+    const [startTime, setStartTime] = useState<Date | null>(isNewInteraction ? null : interactionStartTime);
+    const [endTime, setEndTime] = useState<Date | null>(isNewInteraction ? null : interactionEndTime);
+
+    const formConfig = useMemo(() => getOptionsByPlaceAndSubplaceType(placeType, placeSubType), [placeType, placeSubType]);
+
+    const classes = useStyles();
+    const formClasses = useFormStyles();
+
+    const handleTimeChange = (currentTime: Date, interactionDate: Date, fieldName: string) => {
+        if (isValid(currentTime)) {
+            let newDate = new Date(interactionDate.getTime());
+
+            newDate.setHours(currentTime.getHours());
+            newDate.setMinutes(currentTime.getMinutes());
+
+            if (newDate.getTime()) {
+                clearErrors(fieldName);
+                setValue(fieldName, newDate);
+            }
+        } else {
+            setError(fieldName, {type: 'manual', message: 'שעה לא תקינה'});
+        }
+    }
+
+    const externalizationErrorMessage = useMemo<string>(() => {
+        const initialMessage = '*שים לב כי לא ניתן להחצין מקום אם ';
+        const isPrivatePlace = placeType === placeTypesCodesHierarchy.privateHouse.code;
+        const isTransportationPlace = placeType === placeTypesCodesHierarchy.transportation.code;
+        const externalizationErrors: string[] = [];
+
+        if (isPrivatePlace) {
+            externalizationErrors.push('מדובר בבית פרטי')
+        } else {
```

Comment:
why else? cant all errors happens at the same time?

Reply:
have no idea - was exciting code, just divided into separate components